ocaml/xapi/static_vdis.ml

```diff
@@ -43,23 +43,17 @@ let permanent_vdi_attach ~__context ~vdi ~reason =
 let permanent_vdi_detach ~__context ~vdi =
   info "permanent_vdi_detach: vdi = %s; sr = %s"
     (Ref.string_of vdi) (Ref.string_of (Db.VDI.get_SR ~__context ~self:vdi));
-  Sm.call_sm_vdi_functions ~__context ~vdi
-    (fun srconf srtype sr -> Sm.vdi_detach srconf srtype sr vdi);
+  let vdi_uuid = Db.VDI.get_uuid ~__context ~self:vdi in
   ignore(Helpers.call_script !Xapi_globs.static_vdis
-           [ "del"; Db.VDI.get_uuid ~__context ~self:vdi ])
+           [ "detach"; vdi_uuid ]);
+  ignore(Helpers.call_script !Xapi_globs.static_vdis
+           [ "del"; vdi_uuid ])
```

Comment:
What happens if any of these raises an exception? Is it fine to propagate it without performing `del` for example?

Reply:
I pushed a change to use `log_and_ignore_exn`, that should preserve the original behaviour wrt to exceptions.
According to the comment the  VDI might have already been destroyed, and then detach can fail.